static/js/util.ts

```diff
@@ -158,85 +172,91 @@ export const array_compare = function util_array_compare(a, b) {
  * You must supply a option to the constructor called compute_value
  * which should be a function that computes the uncached value.
  */
-export class CachedValue {
+export class CachedValue<T> {
     _recompute_value = true;
+    _value!: T;
+    compute_value: () => T;
 
-    constructor(opts) {
+    constructor(opts: {compute_value(): T}) {
         this.compute_value = opts.compute_value;
     }
 
-    get() {
+    get(): T {
         if (this._recompute_value) {
             this._value = this.compute_value();
             this._recompute_value = false;
         }
         return this._value;
     }
 
-    reset() {
+    reset(): void {
         this._recompute_value = true;
     }
 }
 
-export function find_wildcard_mentions(message_content) {
-    const mention = message_content.match(/(^|\s)(@\*{2}(all|everyone|stream)\*{2})($|\s)/);
+export function find_wildcard_mentions(message_content: string): string | null {
+    const mention = /(^|\s)(@\*{2}(all|everyone|stream)\*{2})($|\s)/.exec(message_content);
```

Comment:
Why the change from `.match` to `.exec`?

Reply:
This is because of the [prefer-regexp-exec lint rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-regexp-exec.md) for TypeScript files. Added a note about this in the commit message.

Reply:
OK.